KS.Fiks.ASiC-E/AsiceBuilder.cs

```diff
@@ -3,13 +3,58 @@
 using KS.Fiks.ASiC_E.Crypto;
 using KS.Fiks.ASiC_E.Manifest;
 using KS.Fiks.ASiC_E.Model;
+using KS.Fiks.ASiC_E.Sign;
 
 namespace KS.Fiks.ASiC_E;
 
 public sealed class AsiceBuilder : IAsiceBuilder<AsiceArchive>
 {
     private readonly AsiceArchive _asiceArchive;
 
+    private static IManifestCreator FindManifestCreator(
+        ManifestSpec spec,
+        Func<IManifestCreator> makeManifestCreator)
+    => makeManifestCreator != null
+        ? makeManifestCreator() ?? throw new InvalidOperationException(
+            "makeManifestCreator() returned null")
+        : spec switch
+        {
+            // TODO: A standard XAdES manifest has not been implemented
+            // yet, since the initial usecase for XAdES required a
+            // custom manifest format, for which the makeManifestCreator
+            // function parameter was introduced.
+            ManifestSpec.Cades => new CadesManifestCreator(),
+            _ => throw new ArgumentException(
+                "Only CAdES-style manifests are currently supported."),
```

Comment:
Kunne vurdert å bruke NotSupportedException i stedet for ArgumentException, og med en tydeligere melding som peker brukeren i riktig retning. For eksempel at standard XAdES-manifest ikke er implementert, og at man må sende inn en makeManifestCreator når man velger ManifestSpec.Xades.